src/frontend/src/optimizer/rule/batch/batch_filter_expression_simplify_rule.rs

```diff
@@ -0,0 +1,63 @@
+// Copyright 2024 RisingWave Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::collections::HashSet;
+
+use crate::expr::{ExprImpl, ExprType};
+use crate::optimizer::plan_node::LogicalScan;
+use crate::optimizer::rule::Rule;
+use crate::optimizer::{BoxedRule, PlanRef};
+use crate::utils::Condition;
+
+pub struct BatchFilterExpressionSimplifyRule {}
+impl Rule for BatchFilterExpressionSimplifyRule {
+    fn apply(&self, plan: PlanRef) -> Option<PlanRef> {
+        let scan: &LogicalScan = plan.as_logical_scan()?;
+        let predicate = ConditionRewriter::rewrite_condition(scan.predicate().clone());
+        Some(scan.clone_with_predicate(predicate).into())
+    }
+}
+
+impl BatchFilterExpressionSimplifyRule {
+    pub fn create() -> BoxedRule {
+        Box::new(BatchFilterExpressionSimplifyRule {})
+    }
+}
+
+pub struct ConditionRewriter {}
+impl ConditionRewriter {
+    pub fn rewrite_condition(condition: Condition) -> Condition {
+        let conjunctions = condition.conjunctions.clone();
```

Comment:
What happens If we have `a = a` condition?

Reply:
It will be returned without modified, since the target optimize pattern is Not(e) AND (e).

Reply:
But the code doesn't look so.